src/app/authentication/auth.service.ts

```diff
@@ -1,46 +1,81 @@
 import { Injectable } from '@angular/core';
+import * as JWT from 'jwt-decode';
 import { AppComponent } from '../app.component';
 
 declare let Msal: any;
+
 @Injectable()
 export class AuthService {
+
     private app: any;
 
     constructor() {
-        this.app = new Msal.UserAgentApplication((window as any).ClientId, '', this.authCallback, {});
-    }
-
-    public authCallback = (errorDesc, token, error) => {
-        if (token) {
-            localStorage.setItem('status', 'authenticated');
-        } else if (errorDesc || error) {
-            localStorage.setItem('status', 'anonymous');
-        }
+        this.app = new Msal.UserAgentApplication((window as any).ClientId, '',
+            {});
     }
 
     public async login() {
-        return this.app.loginRedirect(this.getScopes());
+        return this.app.loginPopup(this.defaultUserScopes())
+            .then(async () => {
+                return this.getTokenSilent();
+            }, () => {
+                return false;
+            });
```

Comment:
The idiomatic way of using async is to have an await.
Here you're declaring functions as being async but there are no await statements.

Reply:
The code linter deems it as unnecessary to add the wait.